Makefile

```diff
@@ -170,3 +169,32 @@ $(IMAGE_DIRS): %: %/Dockerfile | check-links
 #
 $(EXTERNAL_DEPS): %:
 	docker pull $(subst @,:,$@)
+
+#
+# Targets and variables for creating the dependency graph of the docker images
+# as an image file.
+#
+GVDIR=graphviz
+GVWHOLE=$(GVDIR)/dependency_graph.gv
+DEPENDENCY_GRAPH=dependency_graph.svg
+GVFRAGS=$(addprefix $(GVDIR)/,$(addsuffix .gv.frag,$(IMAGE_DIRS)))
+
+.PHONY: graph
+graph: $(DEPENDENCY_GRAPH)
+
+clean-graph:
+	-rm -r $(GVDIR)
+	-rm -r $(DEPENDENCY_GRAPH)
+
+$(DEPENDENCY_GRAPH): $(GVWHOLE) Makefile
+	dot -T svg $(filter %.gv,$^) > $@
+
+$(GVWHOLE): $(GVFRAGS) Makefile
+	echo "digraph {" >$@
+	echo 'size="14!" pack=true packmode="array2"' >>$@
+	cat $(filter %.gv.frag,$^) >>$@
+	echo "}" >>$@
+
+$(GVFRAGS): $(GVDIR)/%.gv.frag: %/Dockerfile $(DEPEND_SH)
+	-mkdir -p $(dir $@)
```

Comment:
Unless you're militantly opposed, I'm inclined to keep it as-is since it's a little more explicit. 

Incidentally, the GNU make manual claims the $(@D) form is semi-obsolete with GNU make: https://www.gnu.org/software/make/manual/make.html#Automatic-Variables